src/GraphQLParser/ParserContext.Parse.cs

```diff
@@ -1328,22 +1328,20 @@ private GraphQLType ParseType()
             type = ParseNamedType();
         }
 
-        if (!Skip(TokenKind.BANG))
+        if (Skip(TokenKind.BANG))
         {
-            DecreaseDepth();
-            return type;
+            var nonNull = NodeHelper.CreateGraphQLNonNullType(_ignoreOptions);
+
+            nonNull.Type = type;
+            // move comment from wrapped type to wrapping type
+            nonNull.Comments = type.Comments;
+            type.Comments = null;
+            nonNull.Location = GetLocation(start);
+            type = nonNull;
         }
 
-        var nonNull = NodeHelper.CreateGraphQLNonNullType(_ignoreOptions); //TODO: deal with depth
```

Comment:
TODO resolved, nothing to fix here, I just invert condition to align count of `IncreaseDepth` calls with count of `DecreaseDepth` calls.